controllers/cache/redis.go

```diff
@@ -8,16 +8,47 @@ import (
 // RedisReconciler implement the Reconciler interface and lcm.Controller interface.
 type RedisReconciler struct {
 	HarborCluster *goharborv1.HarborCluster
+	Client        k8s.Client
+	Recorder      record.EventRecorder
+	Log           logr.Logger
+	DClient       k8s.DClient
+	Scheme        *runtime.Scheme
+	ExpectCR      *unstructured.Unstructured
+	ActualCR      *unstructured.Unstructured
+	Labels        map[string]string
+	Name          string
+	Namespace     string
 }
 
 // Reconciler implements the reconcile logic of redis service
 func (redis *RedisReconciler) Reconcile() (*lcm.CRStatus, error) {
-	// TODO
+	redis.Labels = redis.NewLabels()
+	redis.Name = redis.GetHarborClusterName()
+	redis.Namespace = redis.GetHarborClusterNamespace()
+
+	crStatus := &lcm.CRStatus{
+		Phase:      lcm.PendingPhase,
+		Properties: nil,
+	}
+
+	crStatus, err := redis.Provision(crStatus)
+	if err != nil {
+		return crStatus, err
+	}
 	return nil, nil
 }
 
-func (redis *RedisReconciler) Provision(spec *goharborv1.HarborCluster) (*lcm.CRStatus, error) {
-	panic("implement me")
+func (redis *RedisReconciler) Provision(crStatus *lcm.CRStatus) (*lcm.CRStatus, error) {
```

Comment:
This is a interface method implement. MUST NOT change the signature of method.

Reply:
ok, I will move CRstatus to RedisReconciler.